### PR TITLE
Introduce common Entity macros and some other cleanup

### PR DESCRIPTION
**Pull request by JaceCear**

### CODE REVIEW

File: include/game/entity.h [hunk 1]
```diff
@@ -67,8 +50,167 @@ u32 sub_800DF38(Sprite *, s32, s32, Player *);
...
+        = TaskCreate(_task, sizeof(eType), _taskPrio, _taskFlags, _taskDtor);           \
+    eType *eName = TaskGetStructPtr(t);                                                 \
+    Sprite *s = &eName->s;                                                              \
+    eName->base.regionX = spriteRegionX;                                                \
+    eName->base.regionY = spriteRegionY;                                                \
+    eName->base.me = me;                                                                \
+    eName->base.spriteX = me->x;                                                        \
+    eName->base.spriteY = spriteY;
+
+#define ENTITY_INIT_2(eType, eName, _task, _taskPrio, _taskFlags, _taskDtor,            \
```
Comment (freshollie) [inline include/game/entity.h:86]: Commented before, but not a fan of these init macros. Yes we can a small amount of code reuse, but for what? It's not complex and the macro makes things more confusing.

Sometimes copy and pasted code is easier to understand and modify than a macro with so many inputs and different uses for different attributes.

The rest of the macros in here are good! :)

File: include/game/game.h [hunk 2]
```diff
@@ -113,7 +113,7 @@ typedef struct {
 } UNK_03005A70; /* 0x30 */
 
 struct UNK_3005A70 {
-    /* 0x00 */ u8 filler0[0xC];
+    /* 0x00 */ u8 filler0[0xC]; // type = SpriteBase?
```
Comment (freshollie) [inline include/game/game.h:116]: Would makes sense! We are yet to find out exactly how this is created though right?

File: include/global.h [hunk 3]
```diff
@@ -173,6 +173,13 @@ typedef void (*VoidFn)(void);
     a = ((u8)b ^ (u8)a) * -1;                                                           \
     b = (u8)b * -1;
 
+#define NEGATE(var)                                                                     \
+    ({                                                                                  \
+        s32 temp = var;                                                                 \
+        var = -temp;                                                                    \
+    })
+#define DIRECT_NEGATE(var) (var = -var;)
```
Comment (freshollie) [inline include/global.h:181]: Do we really need this? I can understand `NEGATE` if we are using it for non matches?

File: src/game/enemies/madillo.c [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+    s->x = TO_WORLD_POS(me->x, spriteRegionX);
+    s->y = TO_WORLD_POS(me->y, spriteRegionY);
+    SET_MAP_ENTITY_INITIALIZED(me);
+
+    SPRITE_INIT_EXCEPT_POS(s, 16, SA2_ANIM_MADILLO, 0, 0x480, 2);
+}
+
+// (99.94%) https://decomp.me/scratch/seFRx
+NONMATCH("asm/non_matching/Task_MadilloMain.inc", void Task_MadilloMain(void))
```
Comment (freshollie) [inline src/game/enemies/madillo.c:49]: Took a look and I can't figure it out :(

File: include/game/entity.h [hunk 5]
```diff
@@ -67,8 +50,167 @@ u32 sub_800DF38(Sprite *, s32, s32, Player *);
...
+        TaskDestroy(gCurTask);                                                          \
+        return;                                                                         \
+    }
+
+#define ENEMY_DESTROY_IF_INVISIBLE_RAW(_enemy, _mapEntity, _sprite, _posX, _posY)       \
+    if (IS_OUT_OF_DISPLAY_RANGE(_posX, _posY)) {                                        \
+        ENEMY_DESTROY_IF_OUT_OF_CAM_RANGE(_enemy, _mapEntity, _sprite);                 \
+    }
+
+#define ENEMY_DESTROY_IF_INVISIBLE(_enemy, _mapEntity, _sprite)                         \
```
Comment (freshollie) [inline include/game/entity.h:211]: `IF_OFF_SCREEN`?

Comment (freshollie) [reply include/game/entity.h:211]: Invisible implies it's an effect?

File: src/game/enemies/madillo.c (hunk 4 above)
Comment (freshollie) [reply src/game/enemies/madillo.c:49]: Oh. Whoops :D https://decomp.me/scratch/OTp1Y

Btw, we should 100% be using `Q_24_8_NEW` as there are loads of places where things always match with it like this.

Comment (freshollie) [reply src/game/enemies/madillo.c:49]: There is a difference with how the compiler treats `<< 8` vs `* 256` in some cases

Comment (JaceCear) [reply src/game/enemies/madillo.c:49]: > 

How often in this project have I said "WTF I tried that!?" yet? ^^
Good job!

File: include/game/entity.h (hunk 5 above)
Comment (JaceCear) [reply include/game/entity.h:218]: Great name!

File: include/global.h (hunk 3 above)
Comment (JaceCear) [reply include/global.h:181]: Kinda, sorta, I guess?
I just noticed that everywhere this way was specifically used and (var *= -1) didn't usually work.
Could rename to NEGATE or remove entirely. One of the things I just left in and waited on your input! :D

File: include/game/entity.h (hunk 1 above)
Comment (JaceCear) [reply include/game/entity.h:86]: Will think about it.
It definitely is less explicit.

**Review by freshollie — APPROVED** [review]

**Pull request merged**